modules/OFStateManager/module/src/gentable_handlers.c

```diff
@@ -1196,31 +1218,47 @@ indigo_core_gentable_entry_resume(
     indigo_cxn_id_t cxn_id;
     of_object_t *obj;
     bool no_async = true;
+    bool free_obj = false;
 
     if (op_ctx == NULL) {
         /* incorrect resume call */
         AIM_LOG_ERROR("%s should have valid op_ctx", __FUNCTION__);
         return;
     }
 
+    /* The connection may have gone. However, we still process the entry
+     * add or delete. The OFConnectionManager will take care the connection
+     * issues.
+     */
     cxn_id = op_ctx->cxn_id;
     obj = op_ctx->obj;
     no_async = op_ctx->no_async;
+    free_obj = op_ctx->free_obj;
     /* if no_async is true, it means that this op_ctx is not dynamic allocated */
     if (op_ctx->no_async == false) {
+        op_ctx->magic = 0;
         aim_free(op_ctx);
+    } else {
+        /* no_async is used for table cleanup. don't expect this call. */
+        AIM_LOG_INFO("%s: internal entry deletion, ignore resume call", __FUNCTION__);
+        return;
```

Comment:
I'm a little confused about this part. What is "internal entry deletion"?

I think the `indigo_core_gentable_entry_resume` can be called:
(1) internally by statemanager, and
(2) externally by driver.

Maybe it's worth adding more comments to describe what you really want to do with each case.

Reply:
As mentioned in another comment, the clear request can be triggered by test script or when the table "uninit". In both case, we don't want to wait for the driver async operation.

Reply:
Ok that makes sense.